src/api/v1/db/oracledb/ingredients-dao.js

```diff
@@ -258,14 +230,40 @@ const updateIngredientById = async (body) => {
 
     if (result.rowsAffected === 0) return null;
 
-    const rawIngredient = convertOutBindsToRawIngredient(result.outBinds);
+    const rawIngredient = convertOutBindsToRawResource(result.outBinds);
     if (rawIngredient.id !== body.data.id) throw new Error('ID returned from database does not match input ID');
     return serializeIngredient(rawIngredient, `ingredients/${body.data.id}`);
   } finally {
     connection.close();
   }
 };
+
+/**
+ * check if a list of ingredient IDs exists in the database
+ *
+ * @param {string[]} ingredientIds a list of ingredient IDs to check in the database
+ * @returns {boolean} returns true if ingredient IDs are valid
+ */
+const checkIngredientsExist = async (ingredientIds) => {
+  const connection = await getConnection();
+  const bindParams = _.reduce(ingredientIds,
+    (binds, curId) => {
+      binds[`id${curId}`] = curId;
+      return binds;
+    }, {});
+  const query = `SELECT Count(*) AS "count" FROM INGREDIENTS WHERE ID IN (${_.keys(bindParams).map((key) => `:${key}`)})`;
+  try {
+    const result = await connection.execute(query, bindParams);
+    if (parseInt(result.rows[0].count, 10) === ingredientIds.length) {
+      return true;
+    }
+    return false;
+  } finally {
+    connection.close();
+  }
+};
```

Comment:
I'm interested in this approach, although I'm concerned it might be somewhat out of scope of this PR, which is already getting pretty huge. I'm wondering if it would make sense to defer this to the next PR, just to simplify the approval process for this one? 

I also would be curious to see what additional value this abstraction adds. I think that ensuring that every dao method uses `finally` is not a huge hurdle to overcome just with code reviews, and that this is definitely a jump in the complexity of the code that might not be super justified if that were the only benefit. Stuff I think we could consider adding here:

* some kind of error handling functionality that simplifies the interface for the request handler functions -- not sure exactly what this looks like
* automatic serialization of the result of the DAO function (is this possible?)
* validation of the serialized result, since we don't do this currently

And so on, curious to hear your thoughts here.

Reply:
From our meeting, this should be a good investigation to do in a separate ticket. We should look at similar usage in other apis, especially the students api. If all goes well with changes to the pizza api, we can bring in the changes to the skeleton.

Reply:
Sounds like a plan. I will get at ticket opened in a sec, and I would like to plan on implementing this in the next instead instead of this one.